src/haptics.h

```diff
@@ -14,9 +15,95 @@ const int EN_OFFSET = 13;
 const int M_RANGE_START = 0;
 const int M_RANGE_END = N_MOTORS;
 
+typedef struct pulse {
+  float freq; // vibration frequency
+  float dur;  // duration in ms
+} pulse;
+
+typedef struct repeat_pulse {
+  pulse* p; // pointer to pulse
+  int repeat_num;
+} repeat_pulse;
+
 void init_haptics();
 
 /* --------- Test & Debug functions ---------- */
 void haptics_test();
 void control_device(int duration, double percent_motor, int num_cycles);
 void actuate_motor(int motor_pin, int duration, double percent_motor);
+void run_haptics(pulse* p, int num_pulses, int num_cycles);
+void actuate_hz(int motor_pin, int duration, double percent_motor, float frequency);
+
+/* --------- Haptic Sample Library ---------- */
```

Comment:
the more I think about this, I think it would be appropriate to move these to their own header called `haptic_profiles.h` or something. The use of static is still 100% appropriate. That new module can import this header (haptics.cpp).